### PR TITLE
feat(console): gitlab, gitlab self hosted - templates

### PR DESCRIPTION
**Pull request by peintnermax**
Bildschirm­foto 2023-03-10 um 16 54 10
Bildschirm­foto 2023-03-10 um 16 55 04

Implements the UI for the gitlab providers

Partial of #4433 

```[tasklist]
### Definition of Ready
- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

**Review by eliobischof — APPROVED** [review]

File: console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    private toast: ToastService,
+    private injector: Injector,
+    private _location: Location,
+    private breadcrumbService: BreadcrumbService,
+  ) {
+    this.form = new FormGroup({
+      name: new FormControl('', []),
+      issuer: new FormControl('', [Validators.required]),
+      clientId: new FormControl('', [Validators.required]),
+      clientSecret: new FormControl('', [Validators.required]),
```
Comment (eliobischof) [inline console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts:58]: let's merge the one with the custom validators first and replace these

File: console/src/app/modules/providers/provider-gitlab/provider-gitlab.component.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    private route: ActivatedRoute,
+    private toast: ToastService,
+    private injector: Injector,
+    private _location: Location,
+    private breadcrumbService: BreadcrumbService,
+  ) {
+    this.form = new FormGroup({
+      name: new FormControl('', []),
+      clientId: new FormControl('', [Validators.required]),
+      clientSecret: new FormControl('', [Validators.required]),
```
Comment (eliobischof) [inline console/src/app/modules/providers/provider-gitlab/provider-gitlab.component.ts:57]: here too

File: console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts (hunk 1 above)
Comment (eliobischof) [reply console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts:58]: I'll update them on #5409

**Review by eliobischof — APPROVED** [review]

File: console/src/app/modules/providers/provider-gitlab/provider-gitlab.component.ts (hunk 2 above)
Comment (peintnermax) [reply console/src/app/modules/providers/provider-gitlab/provider-gitlab.component.ts:57]: I'd prefer to merge this before #5409 as these validators can be changed there too and there are still some things to be fixed 👍

File: console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts (hunk 1 above)
Comment (peintnermax) [reply console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts:58]: I'd prefer to merge this before #5409 as these validators can be changed there too.

File: console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  constructor(
+    private route: ActivatedRoute,
+    private toast: ToastService,
+    private injector: Injector,
+    private _location: Location,
+    private breadcrumbService: BreadcrumbService,
+  ) {
+    this.form = new FormGroup({
+      name: new FormControl('', []),
+      issuer: new FormControl('', [Validators.required]),
```
Comment (eliobischof) [inline console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts:56]: use custom validator

Comment (peintnermax) [reply console/src/app/modules/providers/provider-gitlab-self-hosted/provider-gitlab-self-hosted.component.ts:56]: done

**Review by livio-a — APPROVED** [review]

**Pull request merged**